### PR TITLE
Limit queue length

### PR DESCRIPTION
**Pull request by tristpinsm**
Addressed #19  and #42.

### CODE REVIEW

File: coco/master.py [hunk 1]
```diff
@@ -258,6 +260,10 @@ def _load_config(self, config_path: os.PathLike):
             for path, file in load_state.items():
                 self.state.read_from_file(path, file)
 
+        # Set max queue length
```
Comment (jrs65) [inline coco/master.py]: Ugh (see slack discussion).

File: coco/master.py [hunk 2]
```diff
@@ -52,15 +54,18 @@
 
     with await redis.conn as r:
 
+        # Check if queue is full
+        if QUEUE_LEN > 0 and await r.llen("queue") >= QUEUE_LEN:
+            # Increment dropped request counter
+            await r.incr(f"dropped_counter_{endpoint}")
+
+            return response.json({"reply": "Coco queue is full.", "status": 503}, status=503)
```
Comment (jrs65) [inline coco/master.py]: I debated whether 429 was a better error here, the meaning "Too many requests" is a bit more explicitly what we want, but I think it's really meant to be when a *single* client sends too many requests and gets throttled. So 503 seems good.

**Comment by tristpinsm** [comment]
I'm trying to improve the unit test at the moment by using async requests instead of subprocesses, so can we hold off on merging until that is done?

**Comment by tristpinsm** [comment]
Done now! I also made some changes to the master endpoint to ensure that the queue length is checked and appended to in one operation.

**Comment by nritsche** [comment]
I am going to merge this. Okay with you, @jrs65?

**Comment by jrs65** [comment]
Yup. Good for me.

**Pull request merged**